### ISSUE TITLE
QuantifiedConstraints appears to confuse HLint

### ISSUE THREAD

**Issue by kozross** [ndmitchell/hlint#759]
Consider the following source code:

```haskell
{-# LANGUAGE QuantifiedConstraints #-}

module Foo where

foo :: (forall a . Show a => Show (Maybe a)) => a -> Int
foo = 3
```

GHC (>= 8.6 of course) will happily compile this, but HLint will instead complain:

```
src/Foo.hs:5:49: Error: Parse error
Found:
    module Foo where
    
  > foo :: (forall a . Show a => Show (Maybe a)) => a -> Int
    foo = 3
  

1 hint
```

Version info from ``hlint --version``:

```
HLint v2.2.2, (C) Neil Mitchell 2006-2019
```

*Edit:* Provided a simpler example reproducing this.

**Comment by ndmitchell** [comment]
Thanks for the report. Almost certainly an issue with HSE, and something that we'll fix when #645 has finished landing.

**Comment by cpeikert** [comment]
HSE has been updated to deal with quantified constraints: https://github.com/haskell-suite/haskell-src-exts/pull/427#issuecomment-548641260

Would it be possible to update hlint accordingly, or is #645 the only planned way forward?

**Comment by ndmitchell** [comment]
I just upgraded to the new version of haskell-src-exts, where this code works, so no longer an issue. Happy to take new HSE releases as they come out in the meantime.
